shm/Island.cpp

```diff
@@ -0,0 +1,14 @@
+#include "Island.hpp"
+
+Island::Island(uint16_t positionX, uint16_t positionY)
+    : position_(Coordinates(positionX, positionY)) {
+    store_ = std::make_shared<Store>();
```

Comment:
make it inside initializer_list. But the best way is pass it through constructor. It could be hard to test, when you can't add mock class for Store